src/util/auth.js

```diff
@@ -21,74 +20,65 @@ export const AuthProvider = ({ children }) => {
 
   const navigate = useNavigate();
 
-  const createUser = (user) => {
-    createUserWithEmailAndPassword(auth, user.email, user.password)
-      .then((userCredential) => {
-        // When this promise runs, the user is Signed Up.
-        // userCredential (name is whatever) is a firebase object.
-        const user = userCredential.user;
+  const INITIAL_CREDITS = 100000;
 
-        setUser(user);
+  const createUser = (userInput) => {
+    createUserWithEmailAndPassword(auth, userInput.email, userInput.password)
+      .then((userCredential) => {
+        const randomUserID = uuidv4();
+        const db = getDatabase();
+        const userRef = ref(db, "users/" + randomUserID);
+        set(userRef, {
+          credits: INITIAL_CREDITS,
+        });
+        setUser(userCredential.user);
         setIsLoggedIn(true);
-
-        console.log("user is: ", user);
-        console.log("user credentials is: ", userCredential);
         navigate("markets");
       })
       .catch((error) => {
-        const errorCode = error.code;
-        const errorMessage = error.message;
-
-        console.log("error code is: ", errorCode);
-        console.log("error message is: ", errorMessage);
+        console.error("Error creating user:", error);
       });
   };
 
-  // important to put in user state as props here!
-  const signInUser = (user) => {
```

Comment:
not sure if user can be changed to userInput here because what I was doing is putting the state "user" inside under the hood i think, this one I need to test